packages/utils/src/optionsList.ts

```diff
@@ -2,18 +2,24 @@ import toConstant from "./toConstant";
 import { RJSFSchema, EnumOptionsType } from "./types";
 
 /** Gets the list of options from the schema. If the schema has an enum list, then those enum values are returned. The
- * labels for the options will be the same as the `value`. If the schema has a `oneOf` or `anyOf`, then the value is the
- * list of `const` values from the schema and the label is either the `schema.title` or the value.
+ * labels for the options will be extracted from the non-standard, RJSF-deprecated `enumNames` if it exists, otherwise
+ * the label will be the same as the `value`. If the schema has a `oneOf` or `anyOf`, then the value is the list of
+ * `const` values from the schema and the label is either the `schema.title` or the value.
  *
  * @param schema - The schema from which to extract the options list
  * @returns - The list of options from the schema
  */
 export default function optionsList(
-  schema: RJSFSchema
+  schema: RJSFSchema & { enumNames?: string[] }
 ): EnumOptionsType[] | undefined {
+  if (schema.enumNames) {
+    console.warn(
+      "RJSF: The enumNames property is deprecated and may be removed in a future major release. For more info, see https://react-jsonschema-form.readthedocs.io/en/latest/usage/single/#custom-labels-for-enum-fields"
```

Comment:
Let's be consistent between my new warning and yours? Do you prefer `RJSF` or `WARNING` or maybe neither because the [last warning](https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/src/utils.js#L360) we had didn't do either. Also, I'd suggest removing the second sentence, or at least fixing the link to `stable` instead of `latest`

Reply:
RJSF is probably unnecessary since the console will indicate where the error is coming from. Warning is also unnecessary. I'll remove the link